native/spark-expr/src/bloom_filter/bloom_filter_might_contain.rs

```diff
@@ -0,0 +1,209 @@
+use std::any::Any;
+use std::sync::Arc;
+
+use crate::bloom_filter::spark_bloom_filter::SparkBloomFilter;
+
+#[derive(Debug)]
+pub struct BloomFilterMightContain {
+    signature: Signature,
+    bloom_filter: Option<SparkBloomFilter>,
+}
+
+impl BloomFilterMightContain {
+    pub fn try_new(bloom_filter_expr: Arc<dyn PhysicalExpr>) -> Result<Self> {
+        // early evaluate the bloom_filter_expr to get the actual bloom filter
+        let bloom_filter = evaluate_bloom_filter(&bloom_filter_expr)?;
+        Ok(Self {
+            bloom_filter,
+            signature: Signature::exact(
+                vec![DataType::Binary, DataType::Int64],
+                Volatility::Immutable,
+            ),
+        })
+    }
+}
+
+fn evaluate_bloom_filter(
+    bloom_filter_expr: &Arc<dyn PhysicalExpr>,
+) -> Result<Option<SparkBloomFilter>> {
+    // bloom_filter_expr must be a literal/scalar subquery expression, so we can evaluate it
+    // with an empty batch with empty schema
+    let batch = RecordBatch::new_empty(Arc::new(Schema::empty()));
+    let bloom_filter_bytes = bloom_filter_expr.evaluate(&batch)?;
+    match bloom_filter_bytes {
+        ColumnarValue::Scalar(ScalarValue::Binary(v)) => {
+            Ok(v.map(|v| SparkBloomFilter::from(v.as_slice())))
+        }
+        _ => internal_err!("Bloom filter expression should be evaluated as a scalar binary value"),
+    }
+}
+
+impl ScalarUDFImpl for BloomFilterMightContain {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn name(&self) -> &str {
+        "might_contain"
+    }
+
+    fn signature(&self) -> &Signature {
+        &self.signature
+    }
+
+    fn return_type(&self, _arg_types: &[DataType]) -> Result<DataType> {
+        Ok(DataType::Boolean)
+    }
+
+    fn invoke_with_args(&self, args: ScalarFunctionArgs) -> Result<ColumnarValue> {
+        execute_might_contain(&self.bloom_filter, &args.args)
+    }
+}
+
+fn execute_might_contain(
+    bloom_filter: &Option<SparkBloomFilter>,
+    args: &[ColumnarValue],
+) -> Result<ColumnarValue> {
+    match &args[0] {
+        ColumnarValue::Scalar(ScalarValue::Int64(optional_value)) => {
+            let result = bloom_filter
+                .as_ref()
+                .and_then(|filter| optional_value.map(|value| filter.might_contain_long(value)));
+            Ok(ColumnarValue::Scalar(ScalarValue::Boolean(result)))
+        }
+        ColumnarValue::Array(values_array) => {
+            let values = values_array
+                .as_any()
+                .downcast_ref::<Int64Array>()
+                .ok_or_else(|| {
+                    DataFusionError::Execution(
+                        "Expected values Array to be an Int64Array".to_lowercase(),
+                    )
+                })?;
+
+            bloom_filter
+                .as_ref()
+                .map(|filter| {
+                    Ok(ColumnarValue::Array(Arc::new(
+                        filter.might_contain_longs(values),
```

Comment:
Does this have the mapping inversion issue as with Scalars, and if so do we need a test case that has an array with nulls in it?